### PR TITLE
test: enable back tests for LuaJIT

### PR DESCRIPTION
**Pull request by Buristan**
Since the commit db351d3bc46cf80f4f673129c46a3a4b8be3cadd ("luajit: bump new version"), which introduces CTest as a launcher for LuaJIT's test suites, the LuaJIT tests are not run by Tarantool since `LUAJIT_USE_TEST` is disabled. This patch enables all tests, except ASAN build, due to #10733.

NO_DOC=testing
NO_CHANGELOG=testing
NO_TEST=enable tests back

### CODE REVIEW

File: test/luajit-test-init.lua [hunk 1]
```diff
@@ -137,6 +138,13 @@ assert(type(pairs) == 'function')
 -- test case.
 loaders.override_builtin_disable()
 
+-- Increase the default Lua memory limit.
+--
+-- Some tests in the tarantool-tests suite of LuaJIT assume that
+-- for the GC64 build we have more than 4 GB of Lua memory.
+-- Increase the limit to 128 TiB (LuaJIT maximum for GC64 mode).
+alloc.setlimit(128 * 1024 * 1024 * 1024 * 1024)
```
Comment (ligurio) [inline test/luajit-test-init.lua:146]: Nit: I would set it under `require('ffi').abi('gc64')`.

**Review by ligurio — APPROVED** [review]
LGTM with a minor comment

File: test/luajit-test-init.lua (hunk 1 above)
Comment (Buristan) [reply test/luajit-test-init.lua]: Added:
```diff
diff --git a/test/luajit-test-init.lua b/test/luajit-test-init.lua
index 0eee1999b..b31bec43d 100644
--- a/test/luajit-test-init.lua
+++ b/test/luajit-test-init.lua
@@ -1,6 +1,7 @@
 -- Disable strict for Tarantool.
 require("strict").off()
 
+local ffi = require('ffi')
 local loaders = require('internal.loaders')
 local alloc = require('internal.alloc')
 
@@ -143,7 +144,9 @@ loaders.override_builtin_disable()
 -- Some tests in the tarantool-tests suite of LuaJIT assume that
 -- for the GC64 build we have more than 4 GB of Lua memory.
 -- Increase the limit to 128 TiB (LuaJIT maximum for GC64 mode).
-alloc.setlimit(128 * 1024 * 1024 * 1024 * 1024)
+if ffi.abi("gc64") then
+  alloc.setlimit(128 * 1024 * 1024 * 1024 * 1024)
+end
 
 -- This is workaround introduced for flaky macosx tests reported by
 -- https://github.com/tarantool/tarantool/issues/7058

```

File: test/luajit-test-init.lua [hunk 2]
```diff
@@ -137,6 +139,15 @@ assert(type(pairs) == 'function')
 -- test case.
 loaders.override_builtin_disable()
 
+-- Increase the default Lua memory limit.
+--
+-- Some tests in the tarantool-tests suite of LuaJIT assume that
+-- for the GC64 build we have more than 4 GB of Lua memory.
+-- Increase the limit to 128 TiB (LuaJIT maximum for GC64 mode).
+if ffi.abi("gc64") then
```
Comment (ligurio) [inline test/luajit-test-init.lua:147]: single quotes

Comment (Buristan) [reply test/luajit-test-init.lua]: Replaced.

```diff
diff --git a/test/luajit-test-init.lua b/test/luajit-test-init.lua
index a41425c43..9afcf255c 100644
--- a/test/luajit-test-init.lua
+++ b/test/luajit-test-init.lua
@@ -144,7 +144,7 @@ loaders.override_builtin_disable()
 -- Some tests in the tarantool-tests suite of LuaJIT assume that
 -- for the GC64 build we have more than 4 GB of Lua memory.
 -- Increase the limit to 128 TiB (LuaJIT maximum for GC64 mode).
-if ffi.abi("gc64") then
+if ffi.abi('gc64') then
   alloc.setlimit(128 * 1024 * 1024 * 1024 * 1024)
 end
 
```

Comment (ligurio) [reply test/luajit-test-init.lua]: Thanks!

File: test/luajit-test-init.lua (hunk 1 above)
Comment (ligurio) [reply test/luajit-test-init.lua]: Thanks!

**Review by Totktonada — APPROVED** [review]
No objections from me.

**Pull request merged**